### PR TITLE
update parsing for single response

### PR DESCRIPTION
**Pull request by ros3bud**
@cgrayson, LGTY? 

- updated success and failure response handling for event.threat
- updated test to reflect changes
- updated var names containing parsed body 
- removed email.result
- removed badEmailReason()

### CODE REVIEW

File: src/email.coffee [hunk 1]
```diff
@@ -43,25 +43,26 @@ response = (vars, req, res) ->
   event = {} 
   if res.status == 200
     parsedBody = JSON.parse(res.body)
-    emailResult = parsedBody.emails?[0].result
-    originalEmail = parsedBody.emails?[0].corrected
-    correctedEmail = parsedBody.emails?[0].email
+    emailsArray = parsedBody.emails?[0]
```
Comment (cgrayson) [inline src/email.coffee]: Admittedly an odd thing to come up with a name for, but this isn't an array, so take another stab. :-)

File: src/email.coffee [hunk 2]
```diff
@@ -43,25 +43,26 @@ response = (vars, req, res) ->
...
-      event.outcome = 'failure'
+      event.outcome = "failure"
       event.reason = webbulaError
-    else if badEmailCheck(emailResult) == true
+    else if badEmail(emailResult)
       event.outcome = "failure"
-      event.reason = badEmailResult(emailResult)
+      event.reason = "email address risk level too high"
       event.threat = emailResult
       event.original = originalEmail or correctedEmail
```
Comment (cgrayson) [inline src/email.coffee]: I'm not clear on why this (and line 63) would be set to `correctedEmail`?

Comment (cgrayson) [inline src/email.coffee]: Does Webbula never include a corrected email (`.email` in their response data) when the result is high risk?

File: src/email.coffee [hunk 3]
```diff
@@ -43,25 +43,26 @@ response = (vars, req, res) ->
...
     else
```
Comment (cgrayson) [inline src/email.coffee]: Nit: 3 space indentation vs. 2 everywhere else.

**Comment by cgrayson** [comment]
Let me know if those comments don't make sense. :-) And note that I added to [this one](https://github.com/activeprospect/leadconduit-integration-webbula/pull/1#discussion_r27817701).

**Comment by ros3bud** [comment]
@cgrayson, I've made the changes you suggested. I also added a new test to make sure that an email is corrected even when the result implies risk. LGTY?

File: spec/email_spec.coffee [hunk 4]
```diff
@@ -117,11 +117,32 @@ describe 'Success Response', ->
...
+      email:
+        outcome: "failure"
+        original: "hugotwoa@aol.co"
+        corrected: "hugotwoa@aol.com"
+        threat: "reputation"
+        reason: "email address risk level too high"
+    response = integration.response({}, {}, res)
+    assert.deepEqual expected, response
+  
+
```
Comment (cgrayson) [inline spec/email_spec.coffee]: Didn't notice this at first, but this `it()` is indented too far, and it's making the subsequent `it()`s not register as part of the "Failure Response" `describe()`.

**Comment by cgrayson** [comment]
Other than that test indention comment, LGTM.

File: spec/email_spec.coffee [hunk 5]
```diff
@@ -117,11 +117,32 @@ describe 'Success Response', ->
...
         corrected: "foo@bar.com"
         original: "foo@bar.co"
-        result: "clean"
+        threat: "none"
     response = integration.response({}, {}, res)
     assert.deepEqual expected, response
 
 describe 'Failure Response', ->
+
+    it 'Email should be corrected even when result implies risk', ->
```
Comment (cgrayson) [inline spec/email_spec.coffee]: Re: my previous note, just realized that it's only _this line_ that's indented too far.

**Comment by ros3bud** [comment]
@cgrayson: noted and updated. Thanks!

**Pull request merged**